packages/appsync-modelgen-plugin/src/__tests__/utils/process-connections.test.ts

```diff
@@ -480,86 +466,86 @@ describe('process connection', () => {
   });
 
   describe('GraphQL vNext getConnectedField tests with @primaryKey and @index', () => {
-    let modelMap: CodeGenModelMap;
+    let hasOneModelMap: CodeGenModelMap;
     let v2ModelMap: CodeGenModelMap;
     let v2IndexModelMap: CodeGenModelMap;
 
     beforeEach(() => {
-      const schema = /* GraphQL */ `
-        type Post @model {
-          comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+      const hasOneSchema = /* GraphQL */ `
+        type BatteryCharger @model {
+          powerSource: PowerSource @hasOne
         }
 
-        type Comment @model @key(name: "byPost", fields: ["postID", "content"]) {
-          postID: ID!
-          content: String!
-          post: Post @connection(fields:['postID'])
+        type PowerSource @model {
+          sourceID: ID! @primaryKey
+          amps: Float!
+          volts: Float!
         }
       `;
 
       const v2Schema = /* GraphQL */ `
         type Post @model {
-          comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+          comments: [Comment] @hasMany(fields: ["id"])
         }
         
         type Comment @model {
           postID: ID! @primaryKey(sortKeyFields: ["content"])
           content: String!
-          post: Post @connection(fields:["postID"])
+          post: Post @belongsTo(fields:["postID"])
         }
       `;
 
       const v2IndexSchema = /* graphQL */ `
         type Post @model {
-          comments: [Comment] @connection(keyName: "byContent", fields: ["id"])
+          comments: [Comment] @hasMany(indexName: "byContent", fields: ["id"])
         }
         
         type Comment @model {
           postID: ID! @primaryKey
           content: String! @index(name: "byContent")
-          post: Post @connection(fields: ["postID"])
+          post: Post @belongsTo(fields: ["postID"])
       `;
 
-      modelMap = {
-        Post: {
-          name: 'Post',
+      hasOneModelMap = {
+        BatteryCharger: {
+          name: 'BatteryCharger',
           type: 'model',
           directives: [],
           fields: [
             {
-              type: 'Comment',
+              type: 'PowerSource',
               isNullable: true,
-              isList: true,
-              name: 'comments',
-              directives: [{ name: 'connection', arguments: { keyName: 'byPost', fields: ['id'] } }],
+              isList: false,
+              name: 'powerSource',
+              directives: [{ name: 'hasOne', arguments: { fields: ['sourceID'] } }],
```

Comment:
Yeah I think I'll split this into two test cases, I accidentally let the commented out schema drift from the model map. I needed the fields defined because the primaryKey was set as sourceID instead of defaulting to id